pkg/test/suite/suite.go

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2025 The Falco Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package suite
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/falcosecurity/event-generator/pkg/test/loader"
+)
+
+// TestInfo wraps a test and provides additional information to it, such as the name of the source it belongs and its
+// index within the source.
+type TestInfo struct {
+	// SourceName is the name of the source the test belongs.
+	SourceName string
+	// Index is the index of the test within the source.
+	Index int
+	// Test is the wrapped test.
+	Test *loader.Test
+}
+
+func (t *TestInfo) string() string {
+	return fmt.Sprintf("test (source: %s, index: %d)", t.SourceName, t.Index)
+}
+
+// Suite represents a test suite for a specific rule.
+type Suite struct {
+	// RuleName is the name of the rule the tests are associated with.
+	RuleName string
+	// TestsInfo contains the list of tests belonging to the suite.
+	TestsInfo []*TestInfo
+}
+
+// validateTestNamesUniqueness verifies that each suite's test has a unique name.
+func (s *Suite) validateTestNamesUniqueness() error {
+	testsInfo := make(map[string]*TestInfo)
```

Comment:
We can already allocate the map at creation time since we already know the max length of `testsInfo`

Reply:
You are absolutely right.